src/__tests__/labs/Labs.test.tsx

```diff
@@ -17,68 +14,93 @@ import Patient from '../../shared/model/Patient'
 import Permissions from '../../shared/model/Permissions'
 import { RootState } from '../../shared/store'
 
-const { TitleProvider } = titleUtil
+const { TitleProvider, useTitle } = titleUtil
 const mockStore = createMockStore<RootState, any>([thunk])
 
-describe('Labs', () => {
-  jest.spyOn(titleUtil, 'useUpdateTitle').mockImplementation(() => jest.fn())
-  jest.spyOn(LabRepository, 'findAll').mockResolvedValue([])
-  jest
-    .spyOn(LabRepository, 'find')
-    .mockResolvedValue({ id: '1234', requestedOn: new Date().toISOString() } as Lab)
-  jest
-    .spyOn(PatientRepository, 'find')
-    .mockResolvedValue({ id: '12345', fullName: 'test test' } as Patient)
+const Title = () => {
+  const { title } = useTitle()
 
-  const setup = async (initialEntry: string, permissions: Permissions[] = []) => {
-    const store = mockStore({
-      user: { permissions },
-    } as any)
+  return <h1>{title}</h1>
+}
 
-    let wrapper: any
-    await act(async () => {
-      wrapper = await mount(
-        <Provider store={store}>
-          <MemoryRouter initialEntries={[initialEntry]}>
-            <TitleProvider>
-              <Labs />
-            </TitleProvider>
-          </MemoryRouter>
-        </Provider>,
-      )
-    })
+const expectedLab = {
+  code: 'L-code',
+  id: '1234',
+  patient: '1234',
+  type: 'Type',
+  requestedOn: new Date().toISOString(),
+} as Lab
+const expectedPatient = {
+  fullName: 'fullName',
+  id: '1234',
+} as Patient
 
-    wrapper.update()
-    return { wrapper: wrapper as ReactWrapper }
+const setup = (initialPath: string, permissions: Permissions[]) => {
+  jest.resetAllMocks()
+  jest.spyOn(LabRepository, 'find').mockResolvedValue(expectedLab)
+  jest.spyOn(PatientRepository, 'find').mockResolvedValue(expectedPatient)
+  const store = mockStore({ user: { permissions } } as any)
+
+  return {
```

Comment:
Not a train smash, but just wondering if it is maybe not easier to read the code if the render and return are 2 separate lines of code as done in other tests source files e.g. src/__tests__/incidents/view/ViewIncidentDetails.test.tsx ?

Reply:
We are honestly trying to eke out as much performance and memory saving as possible from the tests, part of that is these `setup()` functions are abstractions on the `renderResult` object anyway so assigning it to a variable just to spread it seemed like a waste of memory. @nobrayner

Reply:
I've been moving towards a single return that spreads the render call.

IMO it is more succinct